index.js

```diff
@@ -16,28 +16,84 @@ const collectionName = 'dogs'
 app.get('/dogs', (req, res) => {
     mongoClient.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true }, (err, client) => {
         if (err) {
-            return res.status(500).json({
-                success: false,
-                message: 'Error connecting to database.',
-                data: []
-            })
+            client.close()
+            return composeJSON(res, 500, false, 'Error connecting to database.', [])
         }
         let db = client.db(dbName)
         let collection = db.collection(collectionName)
         collection.find({}).toArray((err, docs) => {
             if (err) {
-                return res.status(500).json({
-                    success: false,
-                    message: 'Error getting data from database.',
-                    data: []
-                })
+                client.close()
+                return composeJSON(res, 500, false, 'Error getting data from database.', [])
+            } else {
+                client.close()
+                return composeJSON(res, 200, true, 'Data retrieved successfully', docs)
             }
-            res.status(200).json({
-                success: true,
-                message: 'Data has been successfully retrieved and sent.',
-                data: docs
+        })
+    })
+})
+
+app.post('/dogs/:id/wins', (req, res) => {
+    const id = req.params.id
+    const regex = RegExp('[0-9a-f]{24}')
+    if (regex.exec(id) === null) {
+        return composeJSON(res, 400, false, 'Invalid winner ID.', [])
+    }
+    mongoClient.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true }, (err, client) => {
+        if (err) {
```

Comment:
Similar to the get method, we do the same sort of return multiple time with different data. One way to reduce duplication is to set the inputs to the composeJSON function where we currently return it, and then return it once at the end. Again, happy to show you what I mean

Reply:
Think we're going to drop this for now

Reply:
not doing - because promises